server/src/Application/Validation/Validators/ApplicationUser/ApplicationUserValidationExtensions.cs

```diff
@@ -0,0 +1,52 @@
+using FluentValidation;
+using Microsoft.AspNetCore.Identity;
+using StudyZen.Application.Exceptions;
+using StudyZen.Application.Services;
+using StudyZen.Domain.Constraints;
+using StudyZen.Domain.Entities;
+
+namespace StudyZen.Application.Validation;
+
+public static class ApplicationUserValidationExtensions
+{
+    public static IRuleBuilderOptions<T, string?> Username<T>(this IRuleBuilder<T, string?> ruleBuilder)
+    {
+        return ruleBuilder
+            .MustNotBeNullOrWhitespace()
+            .LengthMustNotExceed(ApplicationUserConstraints.UsernameMaxLength);
+    }
+
+    public static IRuleBuilderOptions<T, string?> Email<T>(this IRuleBuilder<T, string?> ruleBuilder, UserManager<ApplicationUser> userManager)
+    {
+        return ruleBuilder
+            .MustNotBeNullOrWhitespace()
+            .LengthMustNotExceed(ApplicationUserConstraints.EmailMaxLength)
+            .EmailAddress()
+            .MustAsync(async (email, cancellationToken) => await IsUniqueEmail(email, userManager, cancellationToken))
+            .WithErrorCode(ValidationErrorCodes.UserAlreadyExists);
+    }
+
+    public static IRuleBuilderOptions<T, string?> FirstName<T>(this IRuleBuilder<T, string?> ruleBuilder)
```

Comment:
Maybe add a rule so that First and Last names can only contain letters?

Reply:
Yeah, we can use here regex for that , I think

Reply:
I guess we shouldn't discriminate people with names like Æ A-12 Musk